python/pyarrow/includes/libarrow.pxd

```diff
@@ -899,6 +899,41 @@ cdef extern from "arrow/ipc/api.h" namespace "arrow::ipc" nogil:
         c_string GetColumnName(int i)
 
 
+cdef extern from "arrow/csv/api.h" namespace "arrow::csv" nogil:
+
+    cdef cppclass CCSVParseOptions" arrow::csv::ParseOptions":
+        unsigned char delimiter
```

Comment:
We should think ahead about how we will handle multi-char delimiters

Reply:
Well... do those appear in the wild?

Reply:
Yep, they do, unfortunately. And regular expressions. this was implemented in continuum's IOPro, for example: https://github.com/ContinuumIO/TextAdapter/blob/53138c2277cdfcf32e127251313d4f77f81050aa/textadapter/core/text_adapter.c#L1575. In pandas, multiline/regex delimiters are implemented using an extremely slow Python parser